### PR TITLE
Add select and scatter 2 lax

### PR DESCRIPTION
**Pull request by jkr26**
Adds select_and_scatter to lax.

In writing an interpreter to rehydrate HLO modules into Jaxprs, as suggested [here](https://github.com/google/jax/discussions/14147), I encountered SelectAndScatter in rehydrating the gradient of a MaxPool (I believe). Reading the documentation for SelectAndScatter, it was not obvious to me how to implement in terms of alternative lax symbols. But found this symbol, and in my experimental implementation this does indeed seem to be the thing I was looking for. So simply opening a PR adding to lax.

### CODE REVIEW

**Comment by jakevdp** [comment]
Thanks - it seems reasonable to add this to the public lax API. A couple additional things we'd need:

- please add the new function to the list of operators in the docs here: http://go/jax-github/blob/main/docs/jax.lax.rst#operators
- please add a test of the new public API. I believe that the XLA implementation of this operator is somewhat incomplete, so you may find that comprehensive tests fail for some corner cases.

**Comment by hawkinsp** [comment]
One thing to note that since this is a maxpool gradient, it already has something of a test here: https://github.com/google/jax/blob/b374080e855deb2af6d7c5f9decfefd547c94e95/tests/lax_autodiff_test.py#L749

If there is a public API it should be tested explicitly though.

**Comment by jakevdp** [comment]
Closing for now, since we're still waiting on tests and documentation.

Feel free to re-open if you're still interested in working on this!
